src/dotnet/commands/dotnet-run/LaunchSettings/ProjectLaunchSettingsProvider.cs

```diff
@@ -11,9 +11,35 @@ internal class ProjectLaunchSettingsProvider : ILaunchSettingsProvider
 
         public string CommandName => CommandNameValue;
 
-        public LaunchSettingsApplyResult TryApplySettings(JObject document, JObject model, ref ICommand command)
+        public LaunchSettingsApplyResult TryApplySettings(JsonElement document, JsonElement model, ref ICommand command)
         {
-            var config = model.ToObject<ProjectLaunchSettingsModel>();
+            var config = new ProjectLaunchSettingsModel();
+            foreach (var property in model.EnumerateObject())
+            {
+                if (string.Equals(property.Name, nameof(ProjectLaunchSettingsModel.CommandLineArgs), StringComparison.OrdinalIgnoreCase))
+                {
+                    config.CommandLineArgs = property.Value.GetString();
+                }
+                else if (string.Equals(property.Name, nameof(ProjectLaunchSettingsModel.LaunchBrowser), StringComparison.OrdinalIgnoreCase))
+                {
+                    config.LaunchBrowser = property.Value.GetBoolean();
+                }
+                else if (string.Equals(property.Name, nameof(ProjectLaunchSettingsModel.LaunchUrl), StringComparison.OrdinalIgnoreCase))
+                {
+                    config.LaunchUrl = property.Value.GetString();
+                }
+                else if (string.Equals(property.Name, nameof(ProjectLaunchSettingsModel.ApplicationUrl), StringComparison.OrdinalIgnoreCase))
+                {
+                    config.ApplicationUrl = property.Value.ToString();
```

Comment:
Intentional that ApplicationUrl is ToString while LaunchUrl is GetString?

Reply:
Nope, good catch, thanks!